### PR TITLE
Implement julia 0.7 iteration protocol

### PR DESCRIPTION
**Pull request by carlobaldassi**
The old start/next/done protocol was deprecated during the beta phase.
(This also adds deps/build.log in .gitignore while at it.)

When this is merged it would be a good idea to tag a new release I guess.

### CODE REVIEW

File: src/HDF5.jl [hunk 1]
```diff
@@ -1144,19 +1150,29 @@ function names(x::HDF5Attributes)
...
+    end
+    function next(parent::Union{HDF5File, HDF5Group}, iter)
+        iter[2] = h5object(h5o_open_by_idx(checkvalid(parent).id, ".", H5_INDEX_NAME, H5_ITER_INC, iter[1]-1, H5P_DEFAULT), parent)
+        iter[1] += 1
+        (iter[2], iter)
+    end
+else
+    function iterate(parent::Union{HDF5File, HDF5Group}, iter = (1,nothing))
+        n, prev_obj = iter
+        prev_obj ≢ nothing && close(prev_obj)
```
Comment (musm) [inline src/HDF5.jl]: this symbol is equivalent to `!(obj === nothing)`?

File: deps/.gitignore [hunk 2]
```diff
@@ -2,3 +2,4 @@ builds
 src
 deps.jl
 downloads
+build.log
```
Comment (musm) [inline deps/.gitignore]: why is this necessary?

**Comment by musm** [comment]
great thanks. We should probably drop 0.6 support henceforth

File: src/HDF5.jl (hunk 1 above)
Comment (garrison) [reply src/HDF5.jl]: yes (or, equivalently, `!==`)
